src/Payments/GPay.res

```diff
@@ -87,8 +95,15 @@ let make = (
       if dict->Js.Dict.get("gpayResponse")->Belt.Option.isSome {
         let metadata = dict->getJsonObjectFromDict("gpayResponse")
         let obj = metadata->getDictFromJson->itemToObjMapper
+        let gPayBody = {
+          let body = PaymentBody.gpayBody(~payObj=obj, ~connectors)
+          !isGuestCustomer &&
+          (list.payment_type === "new_mandate" || list.payment_type === "setup_mandate")
```

Comment:
This same condition is being used everywhere. Could you please move this out in a separate function and use everywhere